### DISCUSSION TITLE
`reload-dir` specified, but default dir(cwd) still been watching

### DISCUSSION THREAD

**Discussion by j1g5awi** [Kludex/uvicorn#1833] [Potential Issue]
Yes, I know it's a normal behavior. But the document said:

> `--reload-dir <path>` - Specify which directories to watch for python file changes. May be used multiple times. If unused, then by default the whole current directory will be watched. If you are running programmatically use reload_dirs=[] and pass a list of strings.

I just want to specify a dir and **DO NOT** use the default dir(cwd), how can I do that?

**Comment by hofrob** [comment]
I'm also running into this. The implementation just doesn't make any sense to me. `uvicorn` looks at the list of directories and if the `cwd` is not in there, it adds it making all my selected settings completely useless.

https://github.com/encode/uvicorn/blob/d43afed1cfa018a85c83094da8a2dd29f656d676/uvicorn/supervisors/watchfilesreload.py#L71

# My command

`uvicorn "app:start" --reload --reload-include "main_package/" --reload-include "some_other_package/" --use-colors --factory`

For weeks I'm trying to get uvicorn to only watch two directories. I tried debugging it step by step but I'm failing. In the output it even tells me, that it will only watch those two directories:

`INFO:     Will watch for changes in these directories: ['/path/to/my/app/main_package', '/path/to/my/app/some_other_package']`

And then it discards this info and `self.reload_dirs` will just contain `/path/to/my/app`.

Docs and code seem to disagree here, please take another look.

**Reply by ronnix** [reply]
Same issue here. I discovered today that when using `watchfiles`, uvicorn will insist on secretly watching the current directory, in addition to the directories you have specified with `--reload-dir`, without telling you so in the log.

https://github.com/encode/uvicorn/blob/master/uvicorn/supervisors/watchfilesreload.py#L68-L72

PS: this issue was already discussed earlier here : https://github.com/encode/uvicorn/discussions/1783

**Comment by sswam** [comment]
I only noticed this because it's causing me to run out of inotify handles. I guess I need to chdir into the directory rather than relying on the --app-dir and --reload-dir options, which do not work as advertised.

This is definitely a real issue not just a "potential issue".

**Comment by robotadam** [comment]
It looks like this is resolved in #2598 .
